pkg/apis/core/types.go

```diff
@@ -3257,12 +3257,36 @@ type NodeSpec struct {
 	DoNotUse_ExternalID string
 }
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
 // NodeConfigSource specifies a source of node configuration. Exactly one subfield must be non-nil.
 type NodeConfigSource struct {
-	metav1.TypeMeta
-	ConfigMapRef *ObjectReference
+	ConfigMap *ConfigMapNodeConfigSource
+}
+
+type ConfigMapNodeConfigSource struct {
+	// Namespace is the metadata.namespace of the referenced ConfigMap.
+	// This field is required in all cases.
+	Namespace string
+
+	// Name is the metadata.name of the referenced ConfigMap.
+	// This field is required in all cases.
+	Name string
+
+	// UID is the metadata.UID of the referenced ConfigMap.
```

Comment:
I updated the comment to provide additional context.

Reply:
New comment text:
```
	// UID is the metadata.UID of the referenced ConfigMap.
	// This field is currently reqired in Node.Spec.
	// TODO(#61643): This field will be forbidden in Node.Spec when #61643 is resolved.
	//               #61643 changes the behavior of dynamic Kubelet config to respect
	//               ConfigMap updates, and thus removes the ability to pin the Spec to a given UID.
	// TODO(#56896): This field will be required in Node.Status when #56896 is resolved.
	//               #63314 (the PR that resolves #56896) adds a structured status to the Node
	//               object for reporting information about the config. This status requires UID
	//               and ResourceVersion, so that it represents a fully-explicit description of
	//               the configuration in use, while (see previous TODO) the Spec will be
	//               restricted to namespace/name in #61643.
```

Reply:
Thanks.